client/src/ui/components/Profile.tsx

```diff
@@ -0,0 +1,218 @@
+  gameTags: string[];
+  username: string;
+  description: string;
+  profilePicture: string;
+  membership: boolean;
+  auth: boolean;
+}
+
+const Profile: React.FC<ProfileProps> = ({
+  profilePicture,
+  username,
+  description,
+  gameTags,
+  membership,
+  auth,
+}) => {
+  // const authenticated = useContext(Authenticated)
+  // Needs to pass a profile through
+  // let authenticated = true;
+
+  const [editDescription, setEditDescription] = useState(description);
+  const [tempDescription, setTempDescription] = useState(description);
+  const [editProfilePic, setEditProfilePic] = useState(profilePicture);
+  const [tempProfilePic, setTempProfilePic] = useState(profilePicture);
+  const [editGameTags, setEditGameTags] = useState(gameTags);
+  const [tempGameTags, setTempGameTags] = useState(gameTags);
+  const [isEditing, setIsEditing] = useState(false);
+
+  const [selectedOption, setSelectedOption] = useState<string[]>(gameTags);
+
+  const imgInput = useRef<HTMLInputElement>(null);
+
+  const handleSelectionChange = (selectedValues: string[]) => {
+    console.log("New dropdown selection: ", [...new Set(selectedValues)]);
+    setSelectedOption([...new Set(selectedValues)]);
+  };
+
+  const colorClasses = [
+    "text-red",
+    "text-blue",
+    "text-green",
+    "text-yellow",
+    "text-pink",
+    "text-orange",
+  ];
+
+  function handleEditClick() {
+    setIsEditing(!isEditing);
+    setTempDescription(editDescription);
+    setTempProfilePic(editProfilePic);
+    setTempGameTags(editGameTags);
+  }
+
+  function handleSaveClick() {
+    // Sends updates to backend
+    setEditDescription(tempDescription);
+    setEditGameTags(selectedOption);
+    setEditProfilePic(tempProfilePic);
+    setIsEditing(!isEditing);
+  }
+
+  function handlePFPClick() {
+    // Updates the profile picture only when editing mode is on
+    if (isEditing) {
+      console.log("PFP is being edited");
+      imgInput.current?.click();
+    }
+  }
+
+  const handleProfilePicChange = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    // Assert event.target as HTMLInputElement
+    const input = event.target as HTMLInputElement;
+
+    // Check if files exist and if the first file is defined
+    if (input.files && input.files[0]) {
+      const file = input.files[0];
+
+      const reader = new FileReader();
+      reader.onloadend = () => {
+        // Make sure to check if reader.result is a string before setting state
+        if (typeof reader.result === "string") {
+          setTempProfilePic(reader.result);
+        }
+      };
+      reader.readAsDataURL(file);
+    } else {
+      console.log("No file chosen or file access error.");
+    }
+  };
+
+  function list() {
+    // Returns master game list
+    return [
+      "Valorant",
+      "League of Legends",
+      "Clash of Clans",
+      "Lethal Company",
+      "Apex",
+      "Fortnite",
+      "Genshin Impact",
+    ];
+  }
+
+  function getColor() {
```

Comment:
Would add another TODO mentioning our plan to assign colours to games in the database